auslib/db.py

```diff
@@ -526,9 +526,26 @@ def forUpdate(self, rowData, changed_by):
         row['timestamp'] = self.getTimestamp()
         return self._insertStatement(**row)
 
-    def getChange(self, change_id, transaction=None):
-        """ Returns the unique change that matches the give change_id """
-        changes = self.select(where=[self.change_id == change_id], transaction=transaction)
+    def getChange(self, change_id, key=None, value=None, data_version=None, transaction=None):
+        """ Returns the unique change that matches the give change_id or
+            combination of data_version and primary key."""
+
+        if change_id is None:
+            for col in self.table.columns:
+                if col.name == key:
+                    pk = col
+            if pk is None:
+                self.debug.log("Invalid primary key name")
+                return None
```

Comment:
This won't handle multipart keys, such as those on the permissions table. I'm also not a fan of passing key and value separately...how about passing a dict that contains the primary key names+values?

Some additional verification would be good as well, such as ensuring the pk names you have match self.baseTable.primary_key.